app/scripts/modules/core/src/cluster/cluster.service.ts

```diff
@@ -273,6 +273,13 @@ export class ClusterService {
     const nameParts = NameUtils.parseServerGroupName(serverGroup.name);
     if (serverGroup.moniker) {
       Object.assign(serverGroup, serverGroup.moniker);
+      if (serverGroup.moniker.cluster == null && (serverGroup.moniker.stack || serverGroup.moniker.detail)) {
```

Comment:
What scenario is this testing for? A code comment might be useful here.

Reply:
I ended up putting the same code in the corresponding backend changes in Clouddriver, so it can be removed from here.

Reply:
The rationale is explained here: https://github.com/spinnaker/clouddriver/pull/3258#discussion_r252848244